spring-kafka/src/test/java/org/springframework/kafka/support/DefaultKafkaHeaderMapperTests.java

```diff
@@ -332,6 +334,115 @@ void inboundJson() {
+		String multiValueWildeCardHeader2 = "test-wildcard-value2";
+		String singleValueHeader = "test-single-value1";
+
+		DefaultKafkaHeaderMapper mapper = new DefaultKafkaHeaderMapper();
+		mapper.setMultiValueHeaderPatterns(multiValueHeader1, multiValueHeader2, "test-wildcard-*");
+
+		Headers rawHeaders = new RecordHeaders();
+		rawHeaders.add(KafkaHeaders.DELIVERY_ATTEMPT, new byte[] { 0, 0, 0, 1 });
+		rawHeaders.add(KafkaHeaders.ORIGINAL_OFFSET, new byte[] { 0, 0, 0, 1 });
+		rawHeaders.add(RetryTopicHeaders.DEFAULT_HEADER_ATTEMPTS, new byte[] { 0, 0, 0, 5 });
+		rawHeaders.add(singleValueHeader, new byte[] { 0, 0, 0, 6 });
+
+		rawHeaders.add(multiValueHeader1, new byte[] { 0, 0, 0, 0 });
+		rawHeaders.add(multiValueHeader1, new byte[] { 0, 0, 0, 1 });
+		rawHeaders.add(multiValueHeader1, new byte[] { 0, 0, 0, 2 });
+		rawHeaders.add(multiValueHeader1, new byte[] { 0, 0, 0, 3 });
+
+		rawHeaders.add(multiValueHeader2, new byte[] { 0, 0, 0, 4 });
+		rawHeaders.add(multiValueHeader2, new byte[] { 0, 0, 0, 5 });
+
+		rawHeaders.add(multiValueWildeCardHeader1, new byte[] { 0, 0, 0, 6 });
+		rawHeaders.add(multiValueWildeCardHeader1, new byte[] { 0, 0, 0, 7 });
+		rawHeaders.add(multiValueWildeCardHeader2, new byte[] { 0, 0, 0, 8 });
+		rawHeaders.add(multiValueWildeCardHeader2, new byte[] { 0, 0, 0, 9 });
+
+		// WHEN
+		Map<String, Object> mappedHeaders = new HashMap<>();
+		mapper.toHeaders(rawHeaders, mappedHeaders);
+
+		// THEN
+		assertThat(mappedHeaders.get(KafkaHeaders.DELIVERY_ATTEMPT)).isEqualTo(1);
+		assertThat(mappedHeaders.get(KafkaHeaders.ORIGINAL_OFFSET)).isEqualTo(new byte[] { 0, 0, 0, 1 });
+		assertThat(mappedHeaders.get(RetryTopicHeaders.DEFAULT_HEADER_ATTEMPTS)).isEqualTo(new byte[] { 0, 0, 0, 5 });
+		assertThat(mappedHeaders.get(singleValueHeader)).isEqualTo(new byte[] { 0, 0, 0, 6 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiHeader1Values = (List<byte[]>) mappedHeaders.get(multiValueHeader1);
+		assertThat(multiHeader1Values).contains(new byte[] { 0, 0, 0, 0 },
+												new byte[] { 0, 0, 0, 1 },
+												new byte[] { 0, 0, 0, 2 },
+												new byte[] { 0, 0, 0, 3 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiHeader2Values = (List<byte[]>) mappedHeaders.get(multiValueHeader2);
+		assertThat(multiHeader2Values).contains(new byte[] { 0, 0, 0, 4 },
+												new byte[] { 0, 0, 0, 5 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiValueWildeCardHeader1Values = (List<byte[]>) mappedHeaders.get(multiValueWildeCardHeader1);
+		assertThat(multiValueWildeCardHeader1Values).contains(new byte[] { 0, 0, 0, 6 },
+															new byte[] { 0, 0, 0, 7 });
+
+		@SuppressWarnings("unchecked")
+		List<byte[]> multiValueWildeCardHeader2Values = (List<byte[]>) mappedHeaders.get(multiValueWildeCardHeader2);
+		assertThat(multiValueWildeCardHeader2Values).contains(new byte[] { 0, 0, 0, 8 },
+															new byte[] { 0, 0, 0, 9 });
```

Comment:
Can we extract all of these “data under test” to variables?
I think the code would be easier to read.